### PR TITLE
Add a campaigns database layout diagram to dev docs

### PR DESCRIPTION
**Pull request by mrnugget**
This adds a database schema/layout-ish diagram to the dev docs. I built this for [this comment](https://github.com/sourcegraph/sourcegraph/pull/8985#issuecomment-604268658) and thought it could be handy to have this in the future.

Screen Shot 2020-03-26 at 08 25 22

### CODE REVIEW

**Comment by unknwon** [comment]
Nice graph! 

Why ChangesetJob talks to everything?

**Comment by mrnugget** [comment]
> Why ChangesetJob talks to everything?

A `ChangesetJob` turns `Patch` into a `Changeset` as part of a `Campaign`. Since `ChangesetJobs` can be retried and report progress/error, we need those associations to 
1. fetch all `ChangesetJob`s belonging to a `Campaign`
2. find which `Patch` lead to which `Changeset`
etc.

**Comment by unknwon** [comment]
> > Why ChangesetJob talks to everything?
> 
> A `ChangesetJob` turns `Patch` into a `Changeset` as part of a `Campaign`. Since `ChangesetJobs` can be retried and report progress/error, we need those associations to
> 
> 1. fetch all `ChangesetJob`s belonging to a `Campaign`
> 2. find which `Patch` lead to which `Changeset`
>    etc.

Thanks, I think one problem we have in all these graphs is we aren't explicit about we are showing control flow or data flow, and kind of mixed together. Or maybe it's intended?

**Comment by mrnugget** [comment]
> we have in all these graphs is we aren't explicit about we are showing control flow or data flow, and kind of mixed together. Or maybe it's intended?

Hmm, I specifically named this "database layout"/"database schema" to make this clear, because I didn't want to add the columns to the diagram.

File: doc/dev/campaigns_database_layout.dot [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    /* } */
+    /*  */
+    /*  */
+    /* patchD -> { patchset2, repoD } [ fillcolor="lightblue" ] */
+    /*  */
+    /* changesetJobD [label="ChangesetJob D", fillcolor="2"] */
+    /* changesetJobD -> { campaign, patchD, changesetD } [ fillcolor="2" ] */
+    /*  */
+    /* campaign -> { patchset2 } [ fillcolor="6", penwidth="2" ] */
+    /* campaign -> { changesetD } [ fillcolor="6" ] */
```
Comment (unknwon) [inline doc/dev/campaigns_database_layout.dot]: Meant to be deleted?

Comment (mrnugget) [reply doc/dev/campaigns_database_layout.dot]: No, it’s there so show off what multiple PatchSets involved in the same Campaign look like.

**Pull request merged**